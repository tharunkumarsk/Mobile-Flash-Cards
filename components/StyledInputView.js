import React from "react";
import { TextInput, StyleSheet } from "react-native";
import Colors from "../constants/Colors";
import PropTypes from "prop-types";

export function StyledInputView({
  text,
  handleChange,
  handleSubmit,
  placeholder
}) {
  return (
    <TextInput
      style={styles.input}
      value={text}
      onChangeText={handleChange}
      placeholder={placeholder}
      autoFocus={true}
      returnKeyType="done"
      onSubmitEditing={handleSubmit}
    />
  );
}
const styles = StyleSheet.create({
  input: {
    borderWidth: 2,
    borderColor: Colors.green,
    backgroundColor: Colors.white,
    paddingLeft: 15,
    paddingRight: 15,
    marginLeft: 40,
    marginRight: 40,
    borderRadius: 5,
    fontSize: 20,
    height: 50,
    marginBottom: 20
  }
});

StyledInputView.propTypes = {
  text: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  placeholder: PropTypes.string
};
