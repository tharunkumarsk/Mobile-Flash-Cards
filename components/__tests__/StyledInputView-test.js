import React from 'react';
import renderer from 'react-test-renderer';

import {StyledInputView}  from '../StyledInputView';

it(`renders correctly the input view`, () => {
  const tree = renderer.create(<StyledInputView
    handleChange={ handleSubmit = () => {}}
    handleSubmit={ handleSubmit = () => {}}
    text = "Deck name"
    >
    Snapshot test!</StyledInputView>).toJSON();

  expect(tree).toMatchSnapshot();
});