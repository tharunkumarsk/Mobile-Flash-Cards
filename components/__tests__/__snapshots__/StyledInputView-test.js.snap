// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the input view 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  onSubmitEditing={[Function]}
  rejectResponderTermination={true}
  returnKeyType="done"
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#00965e",
      "borderRadius": 5,
      "borderWidth": 2,
      "fontSize": 20,
      "height": 50,
      "marginBottom": 20,
      "marginLeft": 40,
      "marginRight": 40,
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
  underlineColorAndroid="transparent"
  value="Deck name"
/>
`;
